homeassistant/components/zha/core/channels/__init__.py

```diff
@@ -5,404 +5,314 @@
-                attr, min_report_int, max_report_int, reportable_change, **kwargs
-            )
-            self.debug(
-                "reporting '%s' attr on '%s' cluster: %d/%d/%d: Result: '%s'",
-                attr_name,
-                self.cluster.ep_attribute,
-                min_report_int,
-                max_report_int,
-                reportable_change,
-                res,
-            )
-        except (zigpy.exceptions.DeliveryError, Timeout) as ex:
-            self.debug(
-                "failed to set reporting for '%s' attr on '%s' cluster: %s",
-                attr_name,
-                self.cluster.ep_attribute,
-                str(ex),
-            )
-
-    async def async_configure(self):
-        """Set cluster binding and attribute reporting."""
-        if not self._zha_device.skip_configuration:
-            await self.bind()
-            if self.cluster.is_server:
-                for report_config in self._report_config:
-                    await self.configure_reporting(
-                        report_config["attr"], report_config["config"]
-                    )
-                    await asyncio.sleep(uniform(0.1, 0.5))
-            self.debug("finished channel configuration")
-        else:
-            self.debug("skipping channel configuration")
-        self._status = ChannelStatus.CONFIGURED
-
-    async def async_initialize(self, from_cache):
-        """Initialize channel."""
-        self.debug("initializing channel: from_cache: %s", from_cache)
-        self._status = ChannelStatus.INITIALIZED
-
-    @callback
-    def cluster_command(self, tsn, command_id, args):
-        """Handle commands received to this cluster."""
-        pass
-
-    @callback
-    def attribute_updated(self, attrid, value):
-        """Handle attribute updates on this cluster."""
-        pass
+        self.zha_device.hass.data[const.DATA_ZHA][component].append(
+            (entity, (unique_id, self.zha_device, channels))
+        )
 
     @callback
-    def zdo_command(self, *args, **kwargs):
-        """Handle ZDO commands on this cluster."""
-        pass
+    def async_send_signal(self, signal: str, *args: Any) -> None:
+        """Send a signal through hass dispatcher."""
+        async_dispatcher_send(self.zha_device.hass, signal, *args)
 
     @callback
-    def zha_send_event(self, cluster, command, args):
+    def zha_send_event(self, event_data: Dict[str, Union[str, int]]) -> None:
         """Relay events to hass."""
-        self._zha_device.hass.bus.async_fire(
+        self.zha_device.hass.bus.async_fire(
             "zha_event",
             {
-                "unique_id": self._unique_id,
-                "device_ieee": str(self._zha_device.ieee),
-                "endpoint_id": cluster.endpoint.endpoint_id,
-                "cluster_id": cluster.cluster_id,
-                "command": command,
-                "args": args,
+                const.ATTR_DEVICE_IEEE: str(self.zha_device.ieee),
```

Comment:
We're losing info here from the events... I don't think this is a good idea as people and the device triggers are using these fields for automations